src/sys/socket/mod.rs

```diff
@@ -1152,6 +1234,18 @@ pub enum ControlMessage<'a> {
     /// page.
     #[cfg(target_os = "linux")]
     TxTime(&'a u64),
+
+    /// Configure DSCP / IP TOS for outgoing v4 packets.
+    ///
+    /// Further information can be found [here](https://en.wikipedia.org/wiki/Differentiated_services).
+    #[cfg(target_os = "linux")]
+    IpTos(&'a i32),
```

Comment:
Seems like we should rename this variant to `Ipv4Tos`, to be consistent with other `Ipv4Xxx` stuff, e.g., `Ipv4Ttl`:)

-----

Well, I just checked the existing `ControlMessage/ControlMessageOwned` and socket options, looks like Nix does not have a convention for this, some variants/types contains `v4` in their name, and some don't. I think we should include `v4` in the name, as Rust std does this to its types, your thoughts on this?

Reply:
I sort of followed what I found in the existing codebase, where most of the v4 stuff is not prefixed with `v4`, while most of the v6 is prefixed with `v6`. I guess the reason is coming from the names in the kernel headers and in the `libc` crate. I like the clarity of having the prefix also for v4 TBH. I would be nice to have a consistent naming in the crate, but it would be a gigantic breaking change, and this is up to the maintainers to decide IMO :)